app-modules/ai/src/Observers/PromptObserver.php

```diff
@@ -37,11 +37,22 @@
 namespace AdvisingApp\Ai\Observers;
 
 use AdvisingApp\Ai\Models\Prompt;
+use App\Models\Authenticatable;
+use Exception;
 
 class PromptObserver
 {
     public function creating(Prompt $prompt): void
     {
         $prompt->user()->associate(auth()->user());
     }
+
+    public function updating(Prompt $prompt): void
+    {
+        if (auth()->user()->id === $prompt->user_id || auth()->user()->hasRole(Authenticatable::SUPER_ADMIN_ROLE)) {
+            return;
+        }
+
+        throw new Exception('You do not have permission to update this prompt.');
+    }
```

Comment:
This is not what I was intending. You would need to first check if there is an authed User.

If there is then check if any of the confidentiality attributes are being changed (dirty) if they are then throw an exception if the user is not someone that can do so.

Given that this is not being done elsewhere and just would have been a nice to have and it is near deployment. Please just remove / revert this.